module/os/linux/zfs/arc_os.c

```diff
@@ -225,19 +225,17 @@ arc_evictable_memory(void)
 }
 
 /*
- * If sc->nr_to_scan is zero, the caller is requesting a query of the
- * number of objects which can potentially be freed.  If it is nonzero,
- * the request is to free that many objects.
- *
- * Linux kernels >= 3.12 have the count_objects and scan_objects callbacks
- * in struct shrinker and also require the shrinker to return the number
- * of objects freed.
- *
- * Older kernels require the shrinker to return the number of freeable
- * objects following the freeing of nr_to_free.
+ * The _count() function returns the number of free-able objects.
+ * The _scan() function returns the number of objects that were freed.
```

Comment:
Yes, it's an arbitrary count that each shrinker callback can define as they like.  It might make sense to change the arc shrinker to be in units of bytes since that's how the other accounting works (in a future PR).